### PR TITLE
TransactionManagers#createInMemory uses synchronous unlocker.

### PR DESCRIPTION
**Pull request by jkozlowski**
## General
**Before this PR**:

In-memory tests use asynchronous unlocker. This is proving problematic for internal usecases, where we now want to guard visibility of certain writes by a new immutable timestamp. Since this is only used for tests, and test behavior should not materially change hopefully, we should switch this here globally.

**After this PR**:

==COMMIT_MSG==
TransactionManagers#createInMemory uses synchronous unlocker.
==COMMIT_MSG==

**Priority**:

**Concerns / possible downsides (what feedback would you like?)**:
* Existing internal usages for internal queue library should not see change in behavior: they used to rely on the immutable timestamp as a lower bound (we unlock the immutable timestamp lock with #tryUnlock). But the code also obviously needs to go as fast as it is able to read and therefore needs to be resilient to the immutable timestamp jumping arbitrarily forward. Therefore, this is not a problem.
* Technically, this is now doing a bit more work synchronously, but I don't think we care so much for tests.

**Is documentation needed?**:

## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:

**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:

**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:

**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:

**Does this PR need a schema migration?**

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:

**What was existing testing like? What have you done to improve it?**:

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:

## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:

**Has the safety of all log arguments been decided correctly?**:

**Will this change significantly affect our spending on metrics or logs?**:

**How would I tell that this PR does not work in production? (monitors, etc.)**:

**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:

**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:

## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:

**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:

**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:

## Development Process
**Where should we start reviewing?**:

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

**Comment by jkozlowski** [comment]
still testing this out

File: atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java [hunk 1]
```diff
@@ -336,6 +336,7 @@ private static TransactionManager createInMemory(
                 .globalMetricsRegistry(new MetricRegistry())
                 .globalTaggedMetricRegistry(DefaultTaggedMetricRegistry.getDefault())
                 .lockAndTimestampServiceFactory(maybeFactory)
+                .defaultTimelockClientFactory((service, _management) -> TimeLockClient.withSynchronousUnlocker(service))
```
Comment (ergo14) [inline atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:339]: This makes is so we do not get batching for some endpoints (freshTs):

```
        RequestBatchingTimestampService timestampService =
                RequestBatchingTimestampService.create(new TimelockServiceErrorDecorator(timelockService));
```

We probably still want this? Saves up on cycles / threads

Comment (jkozlowski) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:339]: I would be surprised if it matters in most tests, where you just have a single thread, period. But maybe for internal skiing product integration tests? Lemme have a quick look

Comment (jkozlowski) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:339]: Yeah, I'd say let's leave this as is? It looks like RequestBatchingTimestampService just operates on #getFreshTimestamps, which I don't imagine happens a lot in tests?

If this turns out to be a problem, we can fix this up, but I'd be very skeptical.

Comment (jkozlowski) [reply atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java:339]: I am though going to write a quick search internally to usages of TransactionManagers#createInMemory, just to make sure this is used the way I think it is

**Review by ergo14 — APPROVED** [review]

**Pull request merged**